### DISCUSSION TITLE
Using Audiveris "without OMR"

### DISCUSSION THREAD

**Discussion by olin256** [Audiveris/audiveris#763] [Q&A] [answered]
Consider the following scenario: I've got a PDF file with sheet music, which I can parse easily with my own tools, resulting in precise information about all the glyphs, barlines and such stuff. In other words, I can skip the (supposedly) "hardest" part of what Audiveris is usually doing roughly up to the HEADS step. However, I'd still like to use Audiveris' capabilities of putting objects together in a meaningful way (stems with heads, beams with stems, dots, slurs…).

What would be the best way to achieve this? Should I try to produce an .omr which would be then loaded into Audiveris?

**Comment by hbitteur** [comment, marked as answer]
As long as you can craft a valid .omr file, Audiveris will accept it as an input for both the engine and the UI.

A .omr file is just a zip archive of a hierarchy of .png and .xml files, for further details you can start with [this section](https://audiveris.github.io/audiveris/_pages/outputs/omr/).

What is not clear in your message is what you mean by "putting objects together". Are you referring to the engine or the user interface?

**Reply by olin256** [reply]
By "putting objects together" I mean both the engine and the UI – that is, I expect engine to do most of the tedious work with me possibly fixing something manually via the interface.

Thanks for the answer, I'll try producing .omr files and see what happens.
